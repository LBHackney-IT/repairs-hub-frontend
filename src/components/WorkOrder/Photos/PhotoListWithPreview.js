import { PhotoProvider, PhotoView } from 'react-photo-view'
import 'react-photo-view/dist/react-photo-view.css'
import PropTypes from 'prop-types'

const PhotoListWithPreview = ({ fileUrls }) => {
  return (
    <PhotoProvider>
      <div className="photoViewList-photoGroupContainer">
        {fileUrls.map((fileUrl) => (
          <div className="photoViewList-photoGroupItem" key={fileUrl}>
            <PhotoView src={fileUrl}>
              <img src={fileUrl} style={{ width: 'auto', height: '150px' }} />
            </PhotoView>
          </div>
        ))}
      </div>
    </PhotoProvider>
  )
}

PhotoListWithPreview.propTypes = {
  fileUrls: PropTypes.arrayOf(PropTypes.string).isRequired,
}

export default PhotoListWithPreview
