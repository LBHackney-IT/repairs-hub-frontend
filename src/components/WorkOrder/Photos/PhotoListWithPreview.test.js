import { render } from '@testing-library/react'
import PhotoListWithPreview from './PhotoListWithPreview'

describe('PhotoListWithPreview component', () => {
  it('renders component', () => {
    const { asFragment } = render(
      <PhotoListWithPreview fileUrls={['url1', 'url2', 'url3']} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
