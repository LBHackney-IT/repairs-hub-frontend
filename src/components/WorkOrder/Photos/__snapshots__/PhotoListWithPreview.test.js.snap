// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoListWithPreview component renders component 1`] = `
<DocumentFragment>
  <div
    class="photoViewList-photoGroupContainer"
  >
    <div
      class="photoViewList-photoGroupItem"
    >
      <img
        src="url1"
        style="width: auto; height: 150px;"
      />
    </div>
    <div
      class="photoViewList-photoGroupItem"
    >
      <img
        src="url2"
        style="width: auto; height: 150px;"
      />
    </div>
    <div
      class="photoViewList-photoGroupItem"
    >
      <img
        src="url3"
        style="width: auto; height: 150px;"
      />
    </div>
  </div>
</DocumentFragment>
`;
